From 93f841ff3d40ac63938ba0f0dfce2ef0594d04d5 Mon Sep 17 00:00:00 2001
From: Sreedhar K <sreedharx.kundella@intel.com>
Date: Tue, 8 Jul 2014 16:24:12 -0400
Subject: [PATCH 2/2] dc_xpwr_battery: load OCV curve for each boot

BZ: 216672

This patch programs the OCV curve for each time
FG parameters are initialized.

Change-Id: I65718e0cf02c1f83896cc521dbdedbda0f5e32bc
Signed-off-by: Sreedhar K <sreedharx.kundella@intel.com>
Signed-off-by: Daniel Leung <daniel.leung@intel.com>
[dl: massaged the original patch to fit; code unchanged.]
---
 drivers/power/dc_xpwr_battery.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/power/dc_xpwr_battery.c b/drivers/power/dc_xpwr_battery.c
index 9e7fbfa..5394cd8 100644
--- a/drivers/power/dc_xpwr_battery.c
+++ b/drivers/power/dc_xpwr_battery.c
@@ -817,7 +817,17 @@ static void pmic_fg_init_config_regs(struct pmic_fg_info *info)
 	if (ret < 0) {
 		dev_warn(&info->pdev->dev, "FG CNTL reg read err!!\n");
 	} else if ((ret & FG_CNTL_OCV_ADJ_EN) && (ret & FG_CNTL_CAP_ADJ_EN)) {
-		dev_info(&info->pdev->dev, "FG data is already initialized\n");
+		dev_info(&info->pdev->dev,
+			 "FG data except the OCV curve is initialized\n");
+		/*
+		 * ocv curve will be set to default values
+		 * at every boot, so it is needed to explicitly write
+		 * the ocv curve data for each boot
+		 */
+		ret = pmic_fg_program_ocv_curve(info);
+		if (ret < 0)
+			dev_err(&info->pdev->dev,
+				"set ocv curve fail:%d\n", ret);
 		pmic_fg_dump_init_regs(info);
 		return;
 	} else {
-- 
1.8.3.2

